plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java

```diff
@@ -195,7 +195,7 @@ public void acceptDataFile(DataFile dataFile, PartitionSpec partitionSpec)
                             Conversions.fromByteBuffer(column.type(), Optional.ofNullable(dataFile.lowerBounds()).map(a -> a.get(id)).orElse(null)));
                     Object upperBound = convertIcebergValueToTrino(column.type(),
                             Conversions.fromByteBuffer(column.type(), Optional.ofNullable(dataFile.upperBounds()).map(a -> a.get(id)).orElse(null)));
-                    Optional<Long> nullCount = Optional.ofNullable(dataFile.nullValueCounts().get(id));
+                    Optional<Long> nullCount = Optional.ofNullable(dataFile.nullValueCounts()).map(nullCounts -> nullCounts.get(id));
```

Comment:
Please add a test to showcase this issue

Reply:
Here is a naive test case showcasing the bug you fixed

```
    @Test
    public void testNoMetrics()
    {
        assertUpdate("create table table_no_metrics_partitioned (name varchar, country varchar) WITH (partitioning = ARRAY['country'])");
        Table table = IcebergUtil.loadIcebergTable(trinoCatalog, tableOperationsProvider, TestingConnectorSession.SESSION,
                new SchemaTableName("tpch", "table_no_metrics_partitioned"));
        // skip metrics
        table.updateProperties()
                .set("write.metadata.metrics.default", "none")
                .commit();
        assertUpdate("INSERT INTO table_no_metrics_partitioned VALUES ('Christoph', 'AT'), (NULL, 'RO')", 2);
        assertEquals(computeActual("SELECT count(*) FROM \"table_no_metrics_partitioned$partitions\" WHERE data IS NOT NULL").getOnlyValue(), 0L);
    }
```

Reply:
nice, this probably would exercise missing min/max as well, right?  IIRC we didn't add a test then

Reply:
I tried this out, but I'm getting an empty Map for min/max/null-counts instead of `null`. I'd have to look at the Iceberg code, but maybe it changed at some point?

Reply:
We may also want to file an Issue to respect `write.metadata.metrics.default` and the other stats tuning properties at some point

Reply:
I did the test on `master` . 
I'm thinking that this is a good reason to handle #11828  in the near future to be covered against regressions that we can't catch when creating content with the latest version of Iceberg.